### PR TITLE
Ensure request task options are keyed with symbols

### PR DESCRIPTION
**Pull request by imtayadeway**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1542661

Replaces https://github.com/ManageIQ/manageiq/issues/16966

### CODE REVIEW

File: app/controllers/api/subcollections/request_tasks.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def request_tasks_query_resource(object)
       def request_tasks_edit_resource(_object, type, id = nil, data = {})
         raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
         request_task = resource_search(id, type, collection_class(:request_tasks))
-        request_task.update_attributes(:options => request_task.options.merge(data['options'] || {}))
+        request_task.update_attributes(:options => request_task.options.merge(data['options'].deep_symbolize_keys || {}))
```
Comment (mkanoor) [inline app/controllers/api/subcollections/request_tasks.rb]: @imtayadeway data['options'] could be missing should the deep symbolize_keys be outside the round brackets?

Comment (mkanoor) [reply app/controllers/api/subcollections/request_tasks.rb]: @imtayadeway we use deep_symbolize_keys here but for request we use symbolize_keys?
https://github.com/ManageIQ/manageiq-api/blob/f257e71ea7c78063659efe2d5bc77c8cc14f659f/lib/services/api/request_parser.rb#L13

Comment (imtayadeway) [reply app/controllers/api/subcollections/request_tasks.rb]: @mkanoor I updated it so it's a bit easier to read/follow

**Comment by chrisarcand** [comment]
Nice work @imtayadeway

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 1009f5ef9f085754cb1502b13f045d0237033f4c
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Fri Feb 9 14:44:04 2018 -0500

    Merge pull request #317 from imtayadeway/deserialize-miqrequests
    
    Ensure request task options are keyed with symbols
    (cherry picked from commit 1e6328ae0c5749854d0576cd4dffaa5d7fc46c21)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1544458
```

**Pull request merged**